### PR TITLE
User/Linux.pm: Don't lock new users

### PR DESCRIPTION
**Pull request by alexie-smirnoff**
Fixes https://github.com/adjust/infrastructure/issues/15741

### CODE REVIEW

**Comment by monsieurp** [comment]
This PR looks good to me but let's wait for @alip's input 👍

File: lib/Rex/User/Linux.pm [hunk 1]
```diff
@@ -158,6 +158,15 @@ sub create_user {
     }
   }
 
+  # don't lock new users
+  if ( ! (
+       exists $data->{crypt_password}
+    || exists $data->{password}
+    || exists $data->{system} ) ) {
```
Comment (alip) [inline lib/Rex/User/Linux.pm:165]: Can you elaborate why we're checking for system here? This means system users will still be created as locked no? Maybe it's a good idea to mention this in the comment section above.

Comment (alexie-smirnoff) [reply lib/Rex/User/Linux.pm:165]: We are checking for system to not alter the current behavior for the system user creation.
Also system users should not be allowed to login eg to run the login shell.

Comment (alip) [reply lib/Rex/User/Linux.pm:165]: I see, thanks for the explanation.

**Comment by alip** [comment]
Looks good to me too.

**Review by monsieurp — APPROVED** [review]
Tested on esh-binhost-1. It's working!

**Pull request merged**